src/sonic-eventd/rsyslog_plugin/main.cpp

```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+#include <unistd.h>
+#include "rsyslog_plugin.h"
+
+void showUsage() {
+    cout << "Usage for rsyslog_plugin: \n" << "options\n"
+        << "\t-r,required,type=string\t\tPath to regex file\n"
+        << "\t-m,required,type=string\t\tYANG module name of source generating syslog message\n"
+        << "\t-h                     \t\tHelp"
+        << endl;
+}
+
+int main(int argc, char** argv) {
+    string regex_path;
+    string module_name;
+    int option_val;
+
+    while((option_val = getopt(argc, argv, "r:m:h")) != -1) {
+        switch(option_val) {
+            case 'r':
+                regex_path = optarg;
+                break;
+            case 'm':
+                module_name = optarg;
+                break;
+            case 'h':
+            case '?':
+            default:
+                showUsage();
+                return 1;
+        }
+    }
+
+    if(regex_path.empty() || module_name.empty()) { // Missing required rc path
+        cerr << "Error: Missing regex_path and module_name." << endl;
+        return 1;
+    }
+    
+    RsyslogPlugin* plugin = new RsyslogPlugin(module_name, regex_path);
```

Comment:
>new [](http://example.com/codeflow?start=28&length=3)

memory leak. Suggest use smart pointer

Reply:
I believe LGTM could catch this kind of critical bug. Suggest you raise a draft PR to public repo, so you will have all PR checkers result, like build/lgtm/coverage.

Reply:
> > new
> 
> memory leak. Suggest use smart pointer

There is no mem leak here, as process exits at that point. Said that, valgrind will not be happy and it is *always* a good idea to make clean exit.

Reply:
Changed to smart pointer.